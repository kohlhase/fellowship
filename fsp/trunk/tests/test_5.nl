Parameter A: bool
!!! The proof is classical; converted to the following LJ + EM proof:
μthesis:A∨¬A.<μx1:A∨¬A.<classic||(μ'x2:A∨¬A.<x2||x1>,μ'th:¬(A∨¬A).<R(μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<L(a)||μ'x2:A∨¬A.<th||x2*_F_>>>||H1>)||μ'x2:A∨¬A.<th||x2*_F_>>)>||thesis>
we need to prove A∨¬A
  we need to prove A∨¬A
    by classic
    by cases: 
      first case:
        by case hypothesis
        we proved A∨¬A (x2)
        by x2
      done
      second case:
        by case hypothesis
        we proved ¬(A∨¬A) (th)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by a
            trivial
            we proved A∨¬A (x2)
            by th
            and by x2
          absurd
        done
        trivial
        we proved A∨¬A (x2)
        by th
        and by x2
      absurd
done
