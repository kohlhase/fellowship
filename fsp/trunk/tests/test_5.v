(* Coq script *) 
Require Import Classical. 
Implicit Arguments classic [P]. 
Parameter A : Prop. 
Theorem em : (A \/ ~A). 
Proof (match classic with or_introl x1 => (let x2:(A \/ ~A) := x1 in x2) | or_intror x1 => (let th:~(A \/ ~A) := x1 in (let x2:(A \/ ~A) := (or_intror _ (fun (H2:A) => (let a:A := H2 in (let x2:(A \/ ~A) := (or_introl _ a) in (match (th x2) with end))))) in (match (th x2) with end))) end). 
