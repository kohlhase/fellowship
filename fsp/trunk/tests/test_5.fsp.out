
    > Current logic: intuitionistic sequent calculus.
      

    > Current logic: classic sequent calculus.
      
fsp < fsp < 
    > A defined.
      
fsp < fsp < 
    > Proof term: 
      μthesis:A∨¬A.<?1||thesis>
      Natural language:
      we need to prove A∨¬A
        .... (1) <======
      done
      
      1 goal yet to prove!
      
      
       |-----  1
      *:A∨¬A
      
fsp < 
    > Proof term: 
      μthesis:A∨¬A.<μth:A∨¬A.<?1.1||th>||thesis>
      Natural language:
      we need to prove A∨¬A
        we need to prove A∨¬A
          .... (1.1) <======
        done
      done
      
      1 goal yet to prove!
      
      
       |-----  1.1
      *:A∨¬A
      th:A∨¬A
      
fsp < 
    > Proof term: 
      μthesis:A∨¬A.<μth:A∨¬A.<R(μH1:¬A.<λH2:A.μH3:⊥.<H2||?1.1.1.1>||H1>)||th>||thesis>
      Natural language:
      we need to prove A∨¬A
        we need to prove A∨¬A
          we need to prove ¬A
            assume A (H2)
            we need to prove ⊥
              by H2
            ...(1.1.1.1) <======
          done
          trivial
        done
      done
      
      1 goal yet to prove!
      
      
      *:A
       |-----  1.1.1.1
      th:A∨¬A
      
fsp < 
    > Closed the last branch: 
      Proof completed!
      
      Proof term: 
      μthesis:A∨¬A.<μth:A∨¬A.<R(μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<L(a)||th>>||H1>)||th>||thesis>
      Natural language:
      !!! The proof is classical; converted to the following LJ + EM proof:
      μthesis:A∨¬A.<μx1:A∨¬A.<classic||(μ'x2:A∨¬A.<x2||x1>,μ'th:¬(A∨¬A).<R(μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<L(a)||μ'x2:A∨¬A.<th||x2*_F_>>>||H1>)||μ'x2:A∨¬A.<th||x2*_F_>>)>||thesis>
      we need to prove A∨¬A
        we need to prove A∨¬A
          by classic
          by cases: 
            first case:
              by case hypothesis
              we proved A∨¬A (x2)
              by x2
            done
            second case:
              by case hypothesis
              we proved ¬(A∨¬A) (th)
              we need to prove ¬A
                assume A (H2)
                we need to prove ⊥
                  by H2
                  we proved A (a)
                  by a
                  trivial
                  we proved A∨¬A (x2)
                  by th
                  and by x2
                absurd
              done
              trivial
              we proved A∨¬A (x2)
              by th
              and by x2
            absurd
      done
      
      
fsp < 
    > em defined.
      
fsp < fsp < 
    > Firing up Coq for confirmation.
      
fsp < 
    > Firing up Coq for confirmation.
      
fsp < 
    > File ./test_5.nl written. Enjoy the reading.
      
fsp < 
    > EOF!

